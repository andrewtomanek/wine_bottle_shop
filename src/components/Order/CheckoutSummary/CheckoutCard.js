import React from "react";
import { connect } from "react-redux";
import classes from "./CheckoutCard.module.css";
import * as actions from "../../../store/actions/index";
import axios from "../../../axios-orders";

const CheckoutCard = ({ itemName, itemType, itemPrice, ...props }) => {
  return (
    <div className={classes.ShopItem}>
      <img
        className={classes.ItemPicture}
        src={`/img/cont/${itemType}.png`}
        alt={itemType}
      />
      <p className={classes.ItemLabel}>{itemName}</p>
      <p className={classes.ItemPrice}>{itemPrice}</p>
      <div className={classes.BuildControl}>
        <button
          className={classes.Less}
          onClick={() => props.onInventoryRemoved(itemType)}
          disabled={props.disabled}
        >
          -
        </button>
        <button
          className={classes.More}
          onClick={() => props.onInventoryAdded(itemType)}
        >
          +
        </button>
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    onInventoryAdded: (ingName) => dispatch(actions.addInventory(ingName)),
    onInventoryRemoved: (ingName) => dispatch(actions.removeInventory(ingName)),
  };
};

export default connect(null, mapDispatchToProps)(CheckoutCard, axios);
