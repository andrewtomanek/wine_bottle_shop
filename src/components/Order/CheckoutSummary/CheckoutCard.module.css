.ItemLabel {
  color: white;
  background-color: var(--orange);
  font-weight: 900;
  font-size: 1.1rem;
  padding: 0.4rem 0.6rem;
  border-radius: 50%;
}

.ItemPrice {
  color: white;
  background-color: var(--orange);
  font-weight: 900;
  font-size: 1.1rem;
  padding: 0.3rem 0.2rem;
  border-radius: 50%;
}

.BuildControl {
  display: grid;
  grid-auto-flow: column;
  margin: 0;
  background-color: var(--red);
  height: 20%;
  z-index: 1;
}

.BuildControl button {
  font: inherit;
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem 0.6rem;
  margin: 0;
  border: 1px solid #aa6817;
  cursor: pointer;
  outline: none;
}

.BuildControl button:disabled {
  background-color: #ac9980;
  border: 1px solid #7e7365;
  color: #ccc;
  cursor: default;
}

.BuildControl button:hover:disabled {
  background-color: #ac9980;
  color: #ccc;
  cursor: not-allowed;
}

.BuildControl .Less {
  background-color: red;
  color: white;
}

.BuildControl .More {
  background-color: var(--green);
  color: white;
}

.BuildControl .Less:hover,
.BuildControl .Less:active {
  background-color: #daa972;
  color: white;
}

.BuildControl .More:hover,
.BuildControl .More:active {
  background-color: #99703f;
  color: white;
}

.ShopItem {
  display: grid;
  justify-content: center;
  align-content: center;
  grid-auto-flow: row;
  width: 90%;
  height: 30vh;
  background-position-x: center;
  background-position-y: center;
  background-size: contain;
  background-repeat: no-repeat;
}

.shopItem1 {
  background-image: url("/img/cont/shopItem1.png");
}

.shopItem2 {
  background-image: url("/img/cont/shopItem2.png");
}
.shopItem3 {
  background-image: url("/img/cont/shopItem3.png");
}
.shopItem4 {
  background-image: url("/img/cont/shopItem4.png");
}
.shopItem5 {
  background-image: url("/img/cont/shopItem5.png");
}
.shopItem6 {
  background-image: url("/img/cont/shopItem6.png");
}
.shopItem7 {
  background-image: url("/img/cont/shopItem7.png");
}
.shopItem8 {
  background-image: url("/img/cont/shopItem8.png");
}
