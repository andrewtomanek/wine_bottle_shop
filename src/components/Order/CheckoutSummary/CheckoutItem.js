import React from "react";
import PropTypes from "prop-types";

import classes from "./CheckoutItem.module.css";

const storeItem = props => {
  let ingredient = null;

  switch (props.type) {
    case "bread-bottom":
      ingredient = <div className={classes.BreadBottom} />;
      break;
    case "bread-top":
      ingredient = (
        <div className={classes.BreadTop}>
          <div className={classes.Seeds1} />
          <div className={classes.Seeds2} />
        </div>
      );
      break;
    case "wine1":
      ingredient = <div className={classes.wine1} />;
      break;
    case "wine2":
      ingredient = <div className={classes.wine2} />;
      break;
    case "wine3":
      ingredient = <div className={classes.wine3} />;
      break;
    case "wine4":
      ingredient = <div className={classes.wine4} />;
      break;
    case "wine5":
      ingredient = <div className={classes.wine5} />;
      break;
    case "wine6":
      ingredient = <div className={classes.wine6} />;
      break;
    case "wine7":
      ingredient = <div className={classes.wine7} />;
      break;
    case "wine8":
      ingredient = <div className={classes.wine8} />;
      break;
    default:
      ingredient = null;
  }

  return ingredient;
};

storeItem.propTypes = {
  type: PropTypes.string.isRequired
};

export default storeItem;
