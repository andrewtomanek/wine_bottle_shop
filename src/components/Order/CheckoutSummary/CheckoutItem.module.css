.ItemLabel {
  display: grid;
  justify-content: center;
  align-content: center;
  color: white;
  font-weight: 900;
  font-size: 1.1rem;
  width: 100%;
  height: 100%;
}

.shopItem1 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem1.png");
}

.shopItem2 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem2.png");
}
.shopItem3 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem3.png");
}
.shopItem4 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem4.png");
}
.shopItem5 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem5.png");
}
.shopItem6 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem6.png");
}
.shopItem7 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem7.png");
}
.shopItem8 {
  background: center / contain no-repeat
    url("../../../assets/images/cont/shopItem8.png");
}
