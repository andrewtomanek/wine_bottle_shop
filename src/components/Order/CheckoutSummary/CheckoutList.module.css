.CheckoutList {
  display: grid;
  grid-template-columns: repeat(6, minmax(15%, 1fr));
  gap: 3rem 1rem;
}

.Empty {
  color: white;
  font-weight: 900;
  font-size: 1.1rem;
  width: 100%;
  height: 100%;
}

.PriceSum {
  background-color: var(--wine);
  color: var(--red);
  font-size: 2rem;
  font-weight: 700;
  padding: 1rem 0.2rem;
  width: 100%;
}

@media (max-width: 600px) {
  .CheckoutList {
    grid-template-columns: repeat(3, minmax(10%, 1fr));
  }
}
