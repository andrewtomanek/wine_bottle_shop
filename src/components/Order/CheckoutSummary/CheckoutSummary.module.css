.CheckoutSummary {
  width: 100vw;
  display: grid;
  grid-gap: 2rem;
  grid-auto-flow: row;
  align-items: center;
  justify-items: center;
  margin: 0;
  padding: 0;
}

.CheckoutBox {
  width: 100vw;
  display: grid;
  align-items: center;
  align-content: space-around;
  justify-content: space-around;
  margin: 0;
  padding: 1rem 0rem;
}

.CheckoutControls {
  width: 50%;
  display: grid;
  grid-gap: 2rem;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  align-content: space-around;
  justify-content: space-around;
}
