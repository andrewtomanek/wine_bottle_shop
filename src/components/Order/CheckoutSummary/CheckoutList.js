import React from "react";

import classes from "./CheckoutList.module.css";
import CheckoutItem from "./CheckoutItem";

const store = props => {
  let transformedItems = Object.keys(props.inventory)
    .map(igKey => {
      return [...Array(props.inventory[igKey])].map((_, i) => {
        return (
          <CheckoutItem
            key={igKey + i}
            type={igKey}
            itemName={props.listItems[igKey]}
          />
        );
      });
    })
    .reduce((arr, el) => {
      return arr.concat(el);
    }, []);

  if (transformedItems.length === 0) {
    transformedItems = <p>Naplň přepravní box</p>;
  }
  return <div className={classes.CheckoutList}>{transformedItems}</div>;
};

export default store;
