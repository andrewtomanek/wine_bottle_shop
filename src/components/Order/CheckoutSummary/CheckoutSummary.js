import React from "react";

import CheckoutList from "./CheckoutList";
import Button from "../../UI/Button/Button";
import classes from "./CheckoutSummary.module.css";

const checkoutSummary = props => {
  return (
    <div className={classes.CheckoutSummary}>
      <div className={classes.CheckoutBox}>
        <CheckoutList
          ingredients={props.ingredients}
          listItems={props.listItems}
        />
      </div>
      <div className={classes.CheckoutControls}>
        <Button btnType="Danger" clicked={props.checkoutCancelled}>
          CANCEL
        </Button>
        <Button btnType="Success" clicked={props.checkoutContinued}>
          CONTINUE
        </Button>
      </div>
    </div>
  );
};

export default checkoutSummary;
