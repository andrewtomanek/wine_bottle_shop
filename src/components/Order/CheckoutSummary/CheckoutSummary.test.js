import React from "react";
import { shallow } from "enzyme";
import CheckoutSummary from "./CheckoutSummary";
import { inventory, listItems, pricesList } from "../../../shared/testData";

const purchaseCancelled = jest.fn();
const purchaseContinued = jest.fn();

const props = {
  inventory,
  listItems,
  pricesList,
  price: 999,
  purchaseCancelled,
  purchaseContinued,
};

describe("Shallow Render CheckoutSummary component", () => {
  it("renders Heading", () => {
    const wrapper = shallow(<CheckoutSummary {...props} />);
    expect(wrapper.find("h4").text()).toContain(
      "Potvdit kompletní obsah objednávky"
    );
  });

  it("renders Danger button", () => {
    const wrapper = shallow(<CheckoutSummary {...props} />);
    expect(wrapper.find({ btnType: "Danger" }).text()).toContain("<button />");
  });

  it("renders Success button", () => {
    const wrapper = shallow(<CheckoutSummary {...props} />);
    expect(wrapper.find({ btnType: "Success" }).text()).toContain("<button />");
  });

  it("renders 2 buttons", () => {
    const wrapper = shallow(<CheckoutSummary {...props} />);
    expect(wrapper.find("button")).toHaveLength(2);
  });

  it("Match Snapshot of CheckoutSummary component", () => {
    const wrapper = shallow(<CheckoutSummary {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
