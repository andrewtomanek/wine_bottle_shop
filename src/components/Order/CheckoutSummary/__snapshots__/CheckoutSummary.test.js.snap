// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render CheckoutSummary component Match Snapshot of CheckoutSummary component 1`] = `
<div
  className="CheckoutSummary"
>
  <div
    className="CheckoutBox"
  >
    <CheckoutList
      inventory={
        Object {
          "shopItem1": 1,
          "shopItem2": 2,
          "shopItem3": 3,
          "shopItem4": 4,
          "shopItem5": 5,
          "shopItem6": 6,
          "shopItem7": 7,
          "shopItem8": 8,
        }
      }
      listItems={
        Object {
          "shopItem1": "Solaris",
          "shopItem2": "Dornfelder",
          "shopItem3": "Agni",
          "shopItem4": "Muscaris",
          "shopItem5": "Sauvignon",
          "shopItem6": "Tramín",
          "shopItem7": "Hibernal",
          "shopItem8": "Pálava",
        }
      }
      pricesList={
        Object {
          "shopItem1": 50,
          "shopItem2": 100,
          "shopItem3": 150,
          "shopItem4": 70,
          "shopItem5": 140,
          "shopItem6": 90,
          "shopItem7": 120,
          "shopItem8": 110,
        }
      }
    />
  </div>
  <div
    className="CheckoutContent"
  >
    <h4
      className="CheckoutHeading"
    >
      Potvdit kompletní obsah objednávky
    </h4>
  </div>
  <div
    className="CheckoutControls"
  >
    <button
      btnType="Danger"
    >
      Zrušit
    </button>
    <button
      btnType="Success"
      disabled={false}
    >
      Pokračovat
    </button>
  </div>
</div>
`;
